vm/chef/cookbooks/singlefs/files/singlefs-install

```diff
@@ -0,0 +1,296 @@
+#!/bin/bash -eu
+#
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Script to configure single node file server based on a Debian image.
+#
+# 1. Create storage pool.
+# 2. Configure NFS and Samba.
+# 3. System configuration: firewall, network settings, etc.
+
+set -x
+
+source /opt/c2d/get_metadata.sh || exit 1
+source /opt/c2d/instance_utils.sh || exit 1
+source /opt/c2d/retry_with_backoff.sh || exit 1
+
+if (( "$(id -u)" > 0 )); then
+  echo "Please run this script as root."
+  exit 1
+fi
+
+# Trigger installation failure
+fail() {
+  echo "${1}"
+  exit 1
+}
+
+# Only install software once.
+readonly COMPLETE_FILE=/opt/c2d/install.complete
+if [[ -f "${COMPLETE_FILE}" ]]; then
+  echo "Software installation already completed."
+  exit 0
+fi
+touch "${COMPLETE_FILE}"
+
+# Redirect all output to the log file.
+readonly LOG_FILE=/opt/c2d/install.log
+exec > >(tee -a "${LOG_FILE}" ) 2>&1
+
+# Metadata inputs from Launcher / Deployment Manager.
+# Mandatory metadata
+readonly STORAGE_POOL="$(get_metadata instance/attributes/STORAGE_POOL_NAME)"
```

Comment:
https://github.com/GoogleCloudPlatform/click-to-deploy/blob/master/vm/chef/cookbooks/c2d-config/files/c2d-utils#L33 - please use current util scripts, in this case - there is a function get_attribute_value.